parser/prism/Translator.cc

```diff
@@ -1408,16 +1416,25 @@ unique_ptr<parser::Hash> Translator::translateHash(pm_node_t *node, pm_node_list
 //
 // This function translates between the two, creating a `Block` node for the given `pm_block_node *`,
 // and wrapping it around the given `Send` node.
-unique_ptr<parser::Node> Translator::translateCallWithBlock(pm_block_node *prismBlockNode,
+unique_ptr<parser::Node> Translator::translateCallWithBlock(pm_node_t *prismBlockOrLambdaNode,
                                                             std::unique_ptr<parser::Node> sendNode) {
-    auto prismParams = prismBlockNode->parameters;
-    auto blockParametersNode = translate(prismParams);
-    auto body = translate(prismBlockNode->body);
+    unique_ptr<parser::Node> parametersNode;
+    unique_ptr<parser::Node> body;
+
+    if (PM_NODE_TYPE_P(prismBlockOrLambdaNode, PM_BLOCK_NODE)) {
+        auto prismBlockNode = reinterpret_cast<pm_block_node *>(prismBlockOrLambdaNode);
+        parametersNode = translate(prismBlockNode->parameters);
+        body = translate(prismBlockNode->body);
+    } else {
+        auto prismLambdaNode = reinterpret_cast<pm_lambda_node *>(prismBlockOrLambdaNode);
```

Comment:
Let's just confirm that `prismBlockOrLambdaNode` really is a block or lambda, and that nothing else can fall into this `else`
 
 
```suggestion
        ENFORCE(PM_NODE_TYPE_P(pair, PM_LAMBDA_NODE))
        
        auto prismLambdaNode = reinterpret_cast<pm_lambda_node *>(prismBlockOrLambdaNode);
```